wallet/concurrent_test.go

```diff
@@ -0,0 +1,40 @@
+package wallet
+
+import (
+	"sync"
+	"testing"
+)
+
+func TestConcurrentSafe(t *testing.T) {
+
+	w := Wallet{}
+
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
```

Comment:
Best practice for using `wg.Add()` is to add the exact amount of goroutines we wait for, at `wg` initialization. So instead of calling `wg.Add(1)` `N`-times you should call `wg.Add(N)` once. (in your case `N` is hardcoded to 10, but it can be a variable also. The main thing is - `N` should be known when you create a `sync.WaitGroup`)

Why is this the best practice?

Because we essentially can have 2 situations:
1. We know the amount of work we're going to be doing;
Ex.: You have a program that reads all files in a folder, and checks if they are all valid `json`. 
```go
folder, _ := os.Folder(..)
files, _ := folder.AllFiles()

var wg sync.WaitGroup
wg.Add(len(files))

for _, f := range files {
    go func(r io.Reader) {
        verifyJSON(r)
        wg.Done()
    }(f)
}
```
2. We don't know the amount of work we're going to be doing.
Ex.: HTTP server. Requests are coming from users and are completely unpredictable.
In this case, `sync.WaitGroup` is bad, using `channels` and launching goroutines as needed will be a much cleaner solution.

So, it is considered bad practice to do `wg.Add(1)`, because to a person who reads your code, it's not immediately clear whether you have situation 1 or 2. And when you're using `wg.Add(n)` it's immediately clear that you have situation 1. Plus, it is a reminder to the programmer, if they start using `WaitGroup` in situation 2, they will see that the only way to do that is with `wg.Add(1)` and think "hmm, I know `wg.Add(1) is bad practice, maybe I'm doing something wrong"